c++11/src/binary_protocol.cpp

```diff
@@ -0,0 +1,58 @@
+#include <lightstep/binary_protocol.h>
+#include <lightstep/tracer.h>
+using namespace opentracing;
+
+namespace lightstep {
+//------------------------------------------------------------------------------
+// Extract
+//------------------------------------------------------------------------------
+Expected<std::unique_ptr<SpanContext>> LightStepBinaryReader::Extract(
+    const Tracer& tracer) const try {
+  auto lightstep_tracer = dynamic_cast<const LightStepTracer*>(&tracer);
+  if (!lightstep_tracer) {
+    return make_unexpected(invalid_carrier_error);
```

Comment:
Instead of failing if the tracer isn't lightstep, could a TextMap encoding be used with the key-values put into the baggage field of BinaryCarrier? It wouldn't be as efficient of course, but that way you could use the BinaryCarrier without locking yourself into the lightstep tracer?